hook.go

```diff
@@ -43,6 +43,25 @@ func NewHook(fileName string) (*Hook, error) {
 	}, nil
 }
 
+// CheckHook checks the given map against HookMapABI to have appropriate values
+// Only checks Type, KeySize, ValueSize
+func CheckHook(m *ebpf.Map) error {
+	abi := m.ABI()
+	if abi.Type != ebpf.UnspecifiedMap && abi.Type != HookMapABI.Type {
```

Comment:
I think checking `ebpf.UnspecifiedMap` made sense in `cilium/ebpf`, but we should always know the type of the map here, so we can always check it.
Maybe just `if abi.Type != HookMapABI.Type {`?

Reply:
fixed